src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -99,14 +102,36 @@ private void parse(String [] argv) {
 		if ( options.has("log_level")) {
 			this.log_level = parseLogLevel((String) options.valueOf("log_level"));
 		}
+
 		if ( options.has("host"))
-			this.mysqlHost = (String) options.valueOf("host");
+			this.maxwellMysql.mysqlHost = (String) options.valueOf("host");
 		if ( options.has("password"))
-			this.mysqlPassword = (String) options.valueOf("password");
+			this.maxwellMysql.mysqlPassword = (String) options.valueOf("password");
 		if ( options.has("user"))
-			this.mysqlUser = (String) options.valueOf("user");
+			this.maxwellMysql.mysqlUser = (String) options.valueOf("user");
 		if ( options.has("port"))
-			this.mysqlPort = Integer.valueOf((String) options.valueOf("port"));
+			this.maxwellMysql.mysqlPort = Integer.valueOf((String) options.valueOf("port"));
+
+		if ( options.has("replication_host"))
+			this.replicationMysql.mysqlHost = (String) options.valueOf("replication_host");
+
+		if ( options.has("replication_password"))
+			this.replicationMysql.mysqlPassword = (String) options.valueOf("replication_password");
+
+		if ( options.has("replication_user"))
+			this.replicationMysql.mysqlUser = (String) options.valueOf("replication_user");
+
+		if ( options.has("replication_port"))
+			this.replicationMysql.mysqlPort = Integer.valueOf((String) options.valueOf("replication_port"));
+
+		if ( this.replicationMysql.mysqlPassword == null && !options.has("replication_password")) {
+			this.replicationMysql = new MaxwellMysqlConfig(this.maxwellMysql.mysqlHost,
+															this.maxwellMysql.mysqlPort,
+															this.maxwellMysql.mysqlUser,
+															this.maxwellMysql.mysqlPassword);
```

Comment:
And also the code where we let the replica fallback to the schema server should probably be in `setDefaults`